### PR TITLE
[MM23348] Selecting an emoji from emoji picker should insert it at the cursor position

### PR DESCRIPTION
**Pull request by clarmso**
#### Summary

This test ensures that the emoji from the emoji picker is inserted in the desired location indicated by the cursor.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/14079

#### Related Pull Requests

- Has server changes - N/A
- Has redux changes - N/A
- Has mobile changes - N/A

#### Screenshots

Here's the expected final result of the test. The grinning emoji is inserted in the middle of "Hello World!"
Screen Shot 2020-04-08 at 6 12 28 PM

Please note that I don't know a way to know if the emoji has been inserted between the words after I hit enter. If you know a good way, please feel free to update the PR.

### CODE REVIEW

File: e2e/cypress/integration/messaging/emoji_insert_position_spec.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    it('M23348 - Selecting an emoji from emoji picker should insert it at the cursor position', () => {
+        cy.get('#post_textbox').type('HelloWorld!');
+        cy.get('#post_textbox').type('{leftarrow}{leftarrow}{leftarrow}{leftarrow}{leftarrow}{leftarrow}');
+        cy.get('#emojiPickerButton').click();
+        cy.get('img[data-testid="grinning"]').click();
+        cy.get('#post_textbox').should('have.value', 'Hello :grinning: World!');
+        cy.get('#post_textbox').type('{enter}');
+        cy.getLastPost().find('p').contains('Hello World!');
+        cy.getLastPost().find('p').children();
+        cy.getLastPost().find('p').find('span[data-emoticon="grinning"]').should('be.visible');
```
Comment (saturninoabril) [inline e2e/cypress/integration/messaging/emoji_insert_position_spec.js:29]: I think this is not accurate to meet "Emoji should be displayed in the post at the position it was inserted". How about something like,
```javascript
cy.getLastPost().find('p').should('have.html', `Hello <span data-emoticon="grinning"><span alt=":grinning:" class="emoticon" title=":grinning:" style="background-image: url(&quot;${Cypress.config('baseUrl')}/static/emoji/1f600.png&quot;);"></span></span> World!`);
```

It's brittle but at least we're sure with the position.

Comment (clarmso) [reply e2e/cypress/integration/messaging/emoji_insert_position_spec.js]: I don't 100% like this solution either, but if you said so I'd put it in.

Comment (saturninoabril) [reply e2e/cypress/integration/messaging/emoji_insert_position_spec.js]: Open to suggestion but can't go on with your initial approach.

Comment (clarmso) [reply e2e/cypress/integration/messaging/emoji_insert_position_spec.js]: I've tried a few things before opening the PR but could not figure out how to ensure that the grinning emoji is between "Hello" and "World". At least we've tried. :)

Comment (saturninoabril) [reply e2e/cypress/integration/messaging/emoji_insert_position_spec.js]: Thanks for trying. We are working on integrating visual regression testing - https://github.com/mattermost/mattermost-webapp/pull/5256., and this is one of those that fits into it. But on the meantime, we're fine as is.

**Pull request merged**